EIPS/eip-2327.md

```diff
@@ -10,25 +10,33 @@ created: 2019-10-28
+The `BEGINDATA` opcode has been suggested before as part of the EIP `Subroutines and Static Jumps for the EVM` [EIP-615](./eip-615.md) as a way to determine the position of jumptables in contract bytecode. It is here introduced in its own right in order to exclude data from the `JUMPDEST` analysis of contracts, making it impossible to jump to data. This makes it easier for static analysis tools to analyse contracts, allows disassemblers, chain explorers and debuggers to not display data as a mess of INVALID opcodes and may even provide a marginal improvement in performance. It also helps scalability because it improves on-chain evaluation of transactions from other chains in that the validation that the code conforms to a certain pattern does not need to do a full jumpdest analysis to see that data is not executed and thus does not have to conform to the pattern (used by the optimism project). Additionally, it paves the way for suggestions such as [EIP-1712](https://github.com/ethereum/EIPs/pull/1712) to disallow unused opcodes, jumptables [EIP-615](./eip-615.md) and speculative proposals to disallow for deployment of contracts with stack usage violations.
 
 ## Specification
-While computing the valid `JUMPDEST`s of a contract, halt analysis if `BEGINDATA` is encountered. A `JUMP` to a value higher than the `PC` value of `BEGINDATA` should throw with a `BAD_JUMP_DESTINATION` error. Bytes past `BEGINDATA` remain accessible via `CODECOPY` and `EXTCODECOPY`.
+While computing the valid `JUMPDEST`s of a contract, halt analysis once the first `BEGINDATA` is encountered. In other words: A jump to a `JUMPDEST` after the first `BEGINDATA` causes a `BAD_JUMP_DESTINATION` error.
 If `BEGINDATA` is encountered during contract execution, it has the same semantics as `STOP`. It uses 0 gas.
 
+Bytes past `BEGINDATA` remain accessible via `CODECOPY` and `EXTCODECOPY`. `BEGINDATA` does not influence `CODESIZE` or `EXTCODESIZE`.
+
 ## Rationale
 The byte `0xb6` was chosen to align with [EIP-615](./eip-615.md).
 The choice to `STOP` if `BEGINDATA` is encountered is somewhat arbitrary. An alternative would be to be to abort the execution with an out-of-gas error.
 
 ## Backwards Compatibility
 The proposal will not change any existing contracts unless their current behaviour relies upon the usage of unused opcodes.
 
+Since contracts have been using data from the very start, in a sense all of them use unused opcodes,
+but they would have to use data in a way that it is skipped during execution and jumped over.
+The Solidity compiler never generated such code. It has to be evaluated whether contracts created by other means
```

Comment:
There are some examples of this at https://github.com/wolflo/meta-jumpers, although I believe the ones in that repo were designed specifically to exhibit this behavior, and does not exactly warrant being regarded as "real world examples"